### ISSUE TITLE
App ANR due to deadlock in appcenter SDK

### ISSUE THREAD

**Issue by R9DCHDC** [microsoft/appcenter-sdk-android#1533]
### **Description**

App ANR due to deadlock in appcenter SDK

### **Repro Steps**

Please list the steps used to reproduce your issue.

1.
2.

### **Details**

1. Which SDK version are you using?
    4.0.0

2. Which OS version did you experience the issue on?
Android 10

3. What device version did you see this error on?  Were you using an emulator or a physical device?
    Huawei

4. What third party libraries are you using?
    Not related

5. Please enable verbose logging for your app using `AppCenter.setLogLevel(Log.VERBOSE)` before your call to `AppCenter.start(...)` and include the logs here:

Main thread got the lock on Distribute and need to acquire the lock on DownloadManagerReleaseDownloader
```
Thread[10]: Id=1 Name="main" State=BLOCKED
com.microsoft.appcenter.distribute.download.manager.DownloadManagerReleaseDownloader.b(Unknown Source:0)
com.microsoft.appcenter.distribute.Distribute.t0(SourceFile:868)
com.microsoft.appcenter.distribute.Distribute.onActivityResumed(SourceFile:513)
android.app.Application.dispatchActivityResumed(Application.java:443)
android.app.Activity.dispatchActivityResumed(Activity.java:1323)
android.app.Activity.onResume(Activity.java:1871)
```

Background thread acquired the lock on DownloadManagerReleaseDownloader and need to acquire the lock on Distribute.
```
Thread[104]: Id=8835 Name="AsyncTask #10" State=BLOCKED  
com.microsoft.appcenter.distribute.Distribute.w0(Unknown Source:0)
com.microsoft.appcenter.distribute.ReleaseDownloadListener.b(SourceFile:65)
com.microsoft.appcenter.distribute.download.manager.DownloadManagerReleaseDownloader.k(SourceFile:151)
com.microsoft.appcenter.distribute.download.manager.DownloadManagerRequestTask.b(SourceFile:49)
com.microsoft.appcenter.distribute.download.manager.DownloadManagerRequestTask.doInBackground(SourceFile:21)
android.os.AsyncTask$3.call(AsyncTask.java:389)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
java.lang.Thread.run(Thread.java:929)
```

**Comment by russelarms** [comment]
@R9DCHDC thanks for reporting this! We'll investigate the problem.
In case you have mappings for these partly obfuscated stack traces, that would help us too.

**Comment by russelarms** [comment]
At the moment, the scenario looks like this:
```
DownloadManagerRequestTask.doInBackground{
DownloadManagerReleaseDownloader.onDownloadStarted{
mListener.onStart(enqueueTime);
}
}
Distribute.getInstance().setDownloading(mReleaseDetails, enqueueTime);
```
```
Distribute.onActivityResumed
Distribute.resumeDistributeWorkflow {
else if (mReleaseDownloader == null || !mReleaseDownloader.isDownloading()) {
    showUpdateDialog();
}
}
```

**Comment by R9DCHDC** [comment]
> At the moment, the scenario looks like this

Yes. That's also what I observed.

**Comment by AnastasiaKubova** [comment]
Hi there! The fix has been released in the [4.3.1 version](https://github.com/microsoft/appcenter-sdk-android/releases/tag/4.3.1) of App Center SDK.
